modules/openapi-generator/src/main/resources/go/model_simple.mustache

```diff
@@ -167,6 +167,18 @@ func (o *{{classname}}) Set{{name}}(v {{vendorExtensions.x-go-base-type}}) {
 	o.{{name}} = v
 {{/isNullable}}
 }
+{{#useDefaultValuesForRequiredVars}}
+{{^isReadOnly}}
+{{#defaultValue}}
+// GetDefault{{baseName}} function assigns the default value {{defaultValue}} to the {{name}} field
+// of the {{classname}} struct and returns the {{{defaultValue}}}.
+func (o *{{classname}}) GetDefault{{baseName}}() interface{}  {
+	o.{{name}} = {{{defaultValue}}}
+	return {{{defaultValue}}}
+}
+{{/defaultValue}}
+{{/isReadOnly}}
+{{/useDefaultValuesForRequiredVars}}
```

Comment:
what about adding a proper cli option for `useDefaultValuesForRequiredVars` : https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java#L128 ?

Reply:
I have made the changes for this.